lib/spitfire.ex

```diff
@@ -2212,7 +2222,7 @@ defmodule Spitfire do
     :eof
   end
 
-  defp peek_token_eat_eol(%{peek_token: {:eol, _token}} = parser) do
```

Comment:
`eat_eol` should be more context aware. Semicolons are not allowed in all contexts or change the meaning